app/models/base.rb

```diff
@@ -307,12 +307,29 @@ def self.parse_datacite_metadata(id: nil, response: nil)
     resource_type_general = attributes.dig("types", "resourceTypeGeneral")
     type = Bolognese::Utils::DC_TO_SO_TRANSLATIONS[resource_type_general.to_s.dasherize] # || attributes.dig("types", "schemaOrg")
 
+    registrant_id = client_id == "crossref.citations" ? get_crossref_member_id(id) : "datacite.#{client_id}"
+
     {
       "@id" => id,
       "@type" => type,
       "datePublished" => get_date(attributes["dates"], "Issued"),
       "proxyIdentifiers" => proxy_identifiers,
-      "registrantId" => "datacite.#{client_id}" }.compact
+      "registrantId" => registrant_id }.compact
+  end
+
+
+  def self.get_crossref_member_id(id, options={})
+    doi = doi_from_url(id)
+    return "crossref.citations" unless doi.present?
+  
+    url = "https://api.crossref.org/works/#{doi}"	
+    # dependency injection
+    response = options[:response] || Maremma.get(url, host: true)	
+    return "crossref.citations" if response.status != 200	
```

Comment:
Why are you returning `crossref.citations`?

Reply:
in case the call to the Crossref API fails (unforeseen reasons) it would keep the event with the crossref.citations registrant_id.  Basically I'm using `crossref.citations` as a default response if no information is obtained. we could query all the events with the missing crossref member_id using  `crossref.citations` label.